src/hooks/useUserAddress.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from "react";
+
+// get connected account address
+function useUserAddress(provider: any) {
```

Comment:
You could useWeb3Modal here to get the provider, instead of passing it. 
Makes it 'cleaner' to use, I guess.